test/unit/ComponentTest.php

```diff
@@ -19,7 +19,7 @@ public function testDependencies() {
     $component = self::$factory->get('test_3');
     $dependencies = $component->getDependencies();
 
-    $this->assertEquals('test_1', $dependencies['test_1']->name);
+    $this->assertEquals('test_1', $dependencies[0]->name);
```

Comment:
This doesn't really test neither the ordering or the uniqueness of the list ;)  ...but I shouldn't really care :D